### ISSUE TITLE
Neo4J GraphQL Apollo Resolver Filter

### ISSUE THREAD

**Issue by marcuskammerlander** [neo4j/graphql#305]
**Is your feature request related to a problem? Please describe.**
Lets Say we have a property in the Schema that is only processed with a custom resolver, call it GesamtSumme. It works great i get back the GesamtSumme.

So now the GesammtSumme isn´t in Neo4J and the directive @ignore is added.

How can i use GesamtSumme for Filtering in GraphQL?

Something like this

...where: {GesamtSumme_GTE: 20.000 })....

I always get back that GesamtSumme_GTE is not know and when i try to extend the input it doesent show up.

So basically it´s a general Question how can i use Custom Resolved fields as an Filter in GraphQL?

**Describe the solution you'd like**
Should be able to use the filter functionalities to resolver too.

**Describe alternatives you've considered**
using a cypher directive

**Comment by darrellwarde** [comment]
This would be extremely difficult, if not impossible, due to the resolution order of resolvers. The `where` argument is resolved first, which is executed as a single Cypher query which returns all fields for the type. The custom resolvers for any additional fields are then resolved, _after_ any filters have already been applied. For this reason I'm tempted to say this is and always will be a no-go, but happy to hear any ideas.

**Comment by danstarns** [comment]
Hey, This is Impossible in my view, the Cypher directive is custom logic and you can return any value. We cant assume anything about what that value is and I have no idea how we could reconcile your arbitrary Cypher to a property on a node.

**Comment by darrellwarde** [comment]
Closing for now, will reopen if someone comes up with some innovative solutions for this!

**Comment by marcuskammerlander** [comment]
I use graphQL-filter (middleware) to achive the UC. It filters the Result of an Query, hopefully it helps somebody

From: Darrell Warde ***@***.***>
Reply to: neo4j/graphql ***@***.***>
Date: Thursday, 22. July 2021 at 10:05
To: neo4j/graphql ***@***.***>
Cc: Marcus Kammerlander ***@***.***>, Author ***@***.***>
Subject: Re: [neo4j/graphql] Neo4J GraphQL Apollo Resolver Filter (#305)


Closing for now, will reopen if someone comes up with some innovative solutions for this!
